tests/tensorflow/test_printing.py

```diff
@@ -79,3 +79,72 @@ def test_unknown_shape():
     expected_out = 'Tensor(Placeholder):0,\tshape=Unknown\t"A:0"\n'
 
     assert std_out.getvalue() == expected_out.lstrip()
+
+
+@pytest.mark.usefixtures("run_with_tensorflow")
+@run_in_graph_mode
+def test_shape_index():
+    """Make sure graph indexing functions as expected."""
+
+    A = tf.compat.v1.placeholder('float', name='A',
+                                 shape=tf.TensorShape([None, None]))
+    x = tf.compat.v1.placeholder('float', name='x',
+                                 shape=tf.TensorShape([None, 1]))
+    y = tf.multiply(1.0, x, name='y')
+
+    z = tf.matmul(A, tf.add(y, y, name='x_p_y'), name='A_dot')
+
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z, depth_lower=3, depth_upper=7)
+
+    expected_out = textwrap.dedent('''
+    ... Op(Add)\t"x_p_y"
+    ... |  Tensor(Mul):0,\tshape=[None, 1]\t"y:0"
+    ... |  |  Op(Mul)\t"y"
+    ... |  |  |  Tensor(Const):0,\tshape=[]\t"y/x:0"
+    ... |  |  |  Tensor(Placeholder):0,\tshape=[None, 1]\t"x:0"
+    ... |  Tensor(Mul):0,\tshape=[None, 1]\t"y:0"
+    ... |  |  Op(Mul)\t"y"
+    ... |  |  |  Tensor(Const):0,\tshape=[]\t"y/x:0"
+    ... |  |  |  Tensor(Placeholder):0,\tshape=[None, 1]\t"x:0"
+    ''')
+
+    assert range_out.getvalue() == expected_out.lstrip()
+
+    range_out = io.StringIO()
+    with redirect_stdout(range_out):        
+        tf_dprint(z)
```

Comment:
This looks like a copy-paste of one of the older tests above.  Did you mean to set `depth_lower` and `depth_upper`?  

This might be a good opportunity to test a *real* cross-section of graph (e.g. maybe `depth_lower=1`, `depth_upper=4`)&mdash;in other words, something that truncates above, below, and across more than one branch at the same level.

Reply:
Updated using the suggested depths. I also removed the second portion of the test since it gets used in a separate test.